src/main/kotlin/com/smallcloud/refactai/code_lens/CodeLensAction.kt

```diff
@@ -34,17 +35,35 @@ class CodeLensAction(
             }.minBy { it.toString().length }
         }
 
+
         return contentMsg
             .replace("%CURRENT_FILE%", relativePath?.toString() ?: filePath.toString())
             .replace("%CURSOR_LINE%", line1.toString())
             .replace("%CODE_SELECTION%", text)
     }
 
+    @OptIn(DelicateCoroutinesApi::class)
```

Comment:
Interesting, is `DelicateCoroutinesApi` OptIn?

Reply:
Is there any other way to implement `setTimeout` functionality? Without that, sometimes chat is not ready yet and file is not getting attached

Reply:
`OptIn` was kindly suggested by IDE when I tried to use `GlobalScope.launch` :)

Reply:
By the way, changed `GlobalScope` to `CoroutineScope`, now we should be fine

Reply:
```kt
Timer().schedule(500) {
    val pos1 = LogicalPosition(line1, 0)
    val pos2 = LogicalPosition(line2, editor.document.getLineEndOffset(line2))

    editor.selectionModel.setSelection(
        editor.logicalPositionToOffset(pos1),
        editor.logicalPositionToOffset(pos2)
    )
}
```

Might work

Reply:
Is `CoroutineScope` undesirable?

Reply:
I think it should be fine since we use coroutines else where :)